src/main/java/edu/ucsb/ucsbcslas/controllers/OnlineOfficeHoursController.java

```diff
@@ -99,4 +130,119 @@
         return ResponseEntity.ok().body(body);
     }
 
+    // @PostMapping(value = "/api/admin/officehours/upload", produces = "application/json")
+    // public ResponseEntity<String> uploadCSV(@RequestParam("csv") MultipartFile csv, @RequestHeader("Authorization") String authorization) throws IOException{
+    //     logger.info("Starting upload CSV");
+    //     AppUser user = authControllerAdvice.getUser(authorization);
+    //     try {
+    //         Reader reader = new InputStreamReader(csv.getInputStream());
+    //         logger.info(new String(csv.getInputStream().readAllBytes()));
+    //         List<OnlineOfficeHours> uploadedOfficeHours = csvToObjectService.parse(reader, OnlineOfficeHours.class);
+    //         List<OnlineOfficeHours> savedOfficeHour = (List<OnlineOfficeHours>) officeHoursRepository.saveAll(uploadedOfficeHours);
+    //         String body = mapper.writeValueAsString(savedOfficeHour);
+    //         return ResponseEntity.ok().body(body);
+    //     } catch(RuntimeException e){
+    //         throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Malformed CSV", e);
+    //     }
+    // }
+
+    @PostMapping(value = "/api/admin/officehours/upload", produces = "application/json")
+    public ResponseEntity<String> uploadCSV(@RequestParam("csv") MultipartFile csv, @RequestHeader("Authorization") String authorization) throws IOException{
+        logger.info("Starting upload CSV");
+        AppUser user = authControllerAdvice.getUser(authorization);
+        if (!authControllerAdvice.getIsAdmin(authorization))
+           return getUnauthorizedResponse("admin");
+        try {
+            BufferedReader csvReader = new BufferedReader(new InputStreamReader(csv.getInputStream()));
+            logger.info(new String(csv.getInputStream().readAllBytes()));
+            
+            String body = "";
+
+            List<OnlineOfficeHours> savedOfficeHour = new ArrayList<>();
+
+            String row = csvReader.readLine();
+            String [] data ;
+            while(row != null){
+                // body += row;
+                row = row.substring(1,row.length() - 1);
+                data = row.split("\",\"");
```

Comment:
I think we may want to modify this regex. I am not sure if this regex properly handles the case when values in the CSV file are not in quotation marks.